plrust/src/allow_list.rs

```diff
@@ -212,6 +212,53 @@ pub enum Error {
     UnsupportedVersionReq(String),
 }
 
+// Tuple describes name of the dependency, version of the dependency, features of the dependecy that are enabled, whether default features enabled
+pub type AllowedDependencyTuple = (String, String, Vec<String>, bool);
+
+// Convert all the allowed dependencies into a vector of tuples
+pub fn get_allowed_dependencies_as_tuple_vector() -> Vec<AllowedDependencyTuple> {
+    let allowlist = load_allowlist().expect("Error loading dependency allow-list");
+    let mut tuples: Vec<AllowedDependencyTuple> = vec![];
+
+    for dependency in allowlist.values() {
+        let mut tuple: Vec<AllowedDependencyTuple> =
+            get_single_dependency_as_tuple_vector(dependency);
+        tuples.append(&mut tuple);
```

Comment:
This could consume the individual tuple instead of using `.append` to move out of it, since we don't use it after this and moving-in is thus legal, using `std::iter::Extend`. This would also allow using anything that returns a correct type when iterated over as the return from `get_single_dependency`, which would include both Vecs and other things.

```suggestion
        tuples.extend(tuple);
```